test/unit/org/apache/cassandra/metrics/TableMetricsTest.java

```diff
@@ -363,6 +363,63 @@ public void testViewMetricsCleanupOnDrop()
         assertEquals(metrics.get().collect(Collectors.joining(",")), 0, metrics.get().count());
     }
 
+    @Test
+    public void testEstimatedPartitionCount() throws InterruptedException
+    {
+        ColumnFamilyStore cfs = recreateTable();
+        assertEquals(0L, cfs.metric.estimatedPartitionCount.getValue().longValue());
+        assertEquals(0L, cfs.metric.estimatedPartitionCountInSSTablesCached.getValue().longValue());
+        long startTime = System.currentTimeMillis();
+
+        int partitionCount = 10;
+        int numRows = 100;
+
+        for (int i = 0; i < numRows; i++)
+            session.execute(String.format("INSERT INTO %s.%s (id, val1, val2) VALUES (%d, '%s', '%s')", KEYSPACE, TABLE, i % partitionCount, "val" + i, "val" + i));
+
+        assertEquals(partitionCount, cfs.metric.estimatedPartitionCount.getValue().longValue());
+        cfs.forceBlockingFlush(ColumnFamilyStore.FlushReason.UNIT_TESTS);
+        assertEquals(partitionCount, cfs.metric.estimatedPartitionCount.getValue().longValue());
+
+        long estimatedPartitionCountInSSTables = cfs.metric.estimatedPartitionCountInSSTablesCached.getValue().longValue();
+        long elapsedTime = System.currentTimeMillis() - startTime;
+        // the caching time is one second; avoid flakiness by only checking if a long time has not passed
+        if (elapsedTime < 980)
```

Comment:
question:
Given the elapsed time calculation happens after calculating `estimatedPartitionCountInSSTables` wouldn't `1000` be as good?

Reply:
Yes, it would. I just don't trust the cache is using the same timer and that they don't tick at different times.

Reply:
Ok, maybe it would be worth adding such clarification in the comment? I am leaving it up to your judgment.